runtime/src/bank.rs

```diff
@@ -3501,6 +3504,12 @@ impl Bank {
         } else if let Some((address, account)) =
             self.check_transaction_for_nonce(tx, next_durable_nonce)
         {
+            let nonce_hash = tx.message().recent_blockhash();
+            if dedup_nonce_lookup.contains(nonce_hash) {
+                return Err(TransactionError::BlockhashNotFound);
+            } else {
+                dedup_nonce_lookup.insert(*nonce_hash);
+            }
```

Comment:
I think this check can make sense to be here for block-validation but not block-production (currently).

Imagine we have 2 txs in a batch, both using same nonce account.
The first one fails due to some **non-recordable** error, 2nd one is a recordable tx.

Block-validation, it doesn't particularly matter if 1st or 2nd one causes an error - the block is invalid if it contains a non-recordable error.
Block-production is a different story though. 1st tx has non-recordable error so it will eventually get dropped; but 2nd tx would get dropped here early on and now neither tx makes it into the block.
With the current separate batches, we'd still have recorded the 2nd tx in a separate batch.

Reply:
make sense, if we remove this check from here then if multiple txs try to use the same nonce and if there are no non-recordable error in the first tx then all the subsequent txs would fail due to `SystemError::NonceBlockhashNotExpired` as the nonce account state is updated for the second tx.

So, the error type would be `SystemError::NonceBlockhashNotExpired` rather than `BlockhashNotFound`

Reply:
I think we should ideally keep the error the same. It's easier to be confident that a tx is still protocol violating vs not if we don't change the error.